resolver.go

```diff
@@ -0,0 +1,45 @@
+package doh
+
+import (
+	"context"
+	"net"
+	"strings"
+
+	madns "github.com/multiformats/go-multiaddr-dns"
+)
+
+type Resolver struct {
+	url string
+}
+
+func NewResolver(url string) *Resolver {
+	if !strings.HasPrefix(url, "https://") {
+		url = "https://" + url
+	}
+	if !strings.HasSuffix(url, "/dns-query") {
+		url = url + "/dns-query"
+	}
```

Comment:
We should remove this, `/dns-query` is only a convention used in examples that epople adopted, but afaik not really required by RFC (`https://firefox.dns.nextdns.io` works without it, this would break use fo this resolver)

Reply:
Ok, I thought it was standard.

Reply:
done.